### PR TITLE
[KYUUBI #2596] Support to control the viewing permissions of databases/tables/views/columns

### PR DESCRIPTION
**Pull request by zhangrenhua**
…columns

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleReplaceShowObjectCommands.scala [hunk 1]
```diff
@@ -17,26 +17,33 @@
...
-import org.apache.kyuubi.plugin.spark.authz.util.{AuthZUtils, ObjectFilterPlaceHolder, WithInternalChild}
+import org.apache.kyuubi.plugin.spark.authz.util.{AuthZUtils, WithInternalChild}
 
 class RuleReplaceShowObjectCommands extends Rule[LogicalPlan] {
   override def apply(plan: LogicalPlan): LogicalPlan = plan match {
     case r: RunnableCommand if r.nodeName == "ShowTablesCommand" => FilteredShowTablesCommand(r)
     // show databases in spark2.4.x
     case r: RunnableCommand if r.nodeName == "ShowDatabasesCommand" =>
       FilteredShowDatabasesCommand(r)
-    case n: LogicalPlan if n.nodeName == "ShowNamespaces" =>
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleReplaceShowObjectCommands.scala:36]: why changing this?

Comment (zhangrenhua) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleReplaceShowObjectCommands.scala]: Yes, no need to changing this file, thank you !

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala [hunk 2]
```diff
@@ -316,6 +316,41 @@ abstract class RangerSparkExtensionSuite extends KyuubiFunSuite with SparkSessio
       doAs("admin", sql(s"DROP DATABASE IF EXISTS $db3"))
     }
   }
+
+  test("KYUUBI #2596 Support to control the viewing permissions of " +
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala:320]: we have already tested databases, and tables?

Can we add a test for columns in an independent test?

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleReplaceShowObjectCommands.scala [hunk 3]
```diff
@@ -37,6 +37,8 @@ class RuleReplaceShowObjectCommands extends Rule[LogicalPlan] {
       ObjectFilterPlaceHolder(n)
     case r: RunnableCommand if r.nodeName == "ShowFunctionsCommand" =>
       FilteredShowFunctionsCommand(r)
+    case r: RunnableCommand if r.nodeName == "ShowColumnsCommand" =>
```
[the remark this answers is not in the document]
Comment (zhangrenhua) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleReplaceShowObjectCommands.scala:40]: okay

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala (hunk 2 above)
Comment (zhangrenhua) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala:320]: okay

**Comment by SteNicholas** [comment]
@zhangrenhua, no any tests?

**Comment by zhangrenhua** [comment]
> @zhangrenhua, no any tests?

use existing test cases?

**Comment by zhangrenhua** [comment]
use existing test cases?

**Comment by zhangrenhua** [comment]
I found that using the "show tables" command will also return the view, how to exclude the view? And spark2.x doesn't support "show views" command @yaooqinn

**Comment by zhangrenhua** [comment]
@SteNicholas Hello, I just get the metadata from the catalog and replace it with the show xxx command. Do I need to add other test cases?
I ran all the test cases and all passed, please help to review the code

**Pull request closed without merging**